src/test/v1.2/advanced/channels-support-1.2.ts

```diff
@@ -0,0 +1,274 @@
+    await fdc3.joinChannel(channel.id);
+    return channel;
+  };
+
+  getSystemChannels = async () => {
+    return await fdc3.getSystemChannels();
+  }
+
+  leaveChannel = async () => {
+    return await fdc3.leaveCurrentChannel();
+  }
+
+  getUserChannel = async (channel: number): Promise<Channel> => {
+    const channels = await fdc3.getSystemChannels();
+    if (channels.length > 0) {
+      return channels[channel - 1];
+    } else {
+      assert.fail("No system channels available for app A");
+    }
+  };
+
+  joinChannel = async (channel: Channel): Promise<void> => {
+    return fdc3.joinChannel(channel.id)
+  }
+
+  createTestChannel = async (): Promise<Channel> => {
+    const channelName = `${this.testAppChannelName}.${this.getRandomId()}`;
+    return fdc3.getOrCreateChannel(channelName);
+  }
+
+  unsubscribeListeners = (): void => {
+    if (listener1 !== undefined) {
+      listener1.unsubscribe();
+      listener1 = undefined;
+    }
+
+    if (listener2 !== undefined) {
+      listener2.unsubscribe();
+      listener2 = undefined;
+    }
+  }
+
+  channelCleanUp = async (): Promise<void> => {
+    this.unsubscribeListeners();
+    await fdc3.leaveCurrentChannel();
+  }
+
+
+  closeChannelsAppWindow = async (testId: string): Promise<void> => {
+    //Tell ChannelsApp to close window
+    const appControlChannel = await broadcastAppChannelCloseWindow(testId);
+
+    //Wait for ChannelsApp to respond
+    await waitForContext("windowClosed", testId, appControlChannel);
+    await wait(constants.WindowCloseWaitTime);
+  }
+
+
+
+  initCompleteListener = async (testId: string): Promise<Context> => {
+    return waitForContext(
+      "executionComplete",
+      testId,
+      await fdc3.getOrCreateChannel("app-control")
+    );
+  }
+
+  openChannelApp = async (testId: string, channelId: string | undefined, commands: string[], historyItems: number = undefined, notify: boolean = true): Promise<void> => {
+    const channelsAppConfig: ChannelsAppConfig = {
+      fdc3ApiVersion: "1.2",
+      testId: testId,
+      channelId,
+      notifyAppAOnCompletion: notify,
+    };
+
+    if (historyItems) {
+      channelsAppConfig.historyItems = historyItems;
+    }
+
+    //Open ChannelsApp then execute commands in order
+    await fdc3.open(
+      "ChannelsApp",
+      buildChannelsAppContext(commands, channelsAppConfig)
+    );
+  }
+
+  setupAndValidateListener1 = (channel: Channel, expectedContextType: string, errorMessage: string, onComplete: (ctx: Context) => void): void => {
+    if (channel) {
+      listener1 = channel.addContextListener(null, (context) => {
+        expect(context.type).to.be.equals(expectedContextType, errorMessage);
+        onComplete(context);
+      });
+    } else {
+      listener1 = fdc3.addContextListener(null, (context) => {
+        expect(context.type).to.be.equals(expectedContextType, errorMessage);
+        onComplete(context);
+      });
+    }
+
+    validateListenerObject(listener1);
+  }
```

Comment:
```suggestion
  setupAndValidateListener1 = (channel: Channel, listenerContextType: string, expectedContextType: string, errorMessage: string, onComplete: (ctx: Context) => void): void => {
    if (channel) {
      listener1 = channel.addContextListener(listenerContextType, (context) => {
        expect(context.type).to.be.equals(expectedContextType, errorMessage);
        onComplete(context);
      });
    } else {
      listener1 = fdc3.addContextListener(listenerContextType, (context) => {
        expect(context.type).to.be.equals(expectedContextType, errorMessage);
        onComplete(context);
      });
    }

    validateListenerObject(listener1);
  }
```

Would require all calls to it to be updated to to specify the listener type (specific type or null). 

next function needs a similar change